### ISSUE TITLE
Plugin not working properly on iOS

### ISSUE THREAD

**Issue by EricHier** [ObjectivityLtd/beacon_monitoring#16]
After the new release, the iOS problem persists. I found that it is still not working properly. 

Once after letting the app run for over one minute, beacons were ranged on one iPhone. Using a different device, there was no way to get it running. Is there a possibility to create debug logs to understand the issue?

**Comment by RazaGR** [comment]
@EricHier were you able to make it work with iOS? I am facing same issue,  @MadDaelim any update for this?

**Comment by EricHier** [comment]
I was not able to make it work reliability on iOS. Help would be highly appreciated 🙏

**Comment by wwwdata** [comment]
What exactly are your issues? I also tested this plugin on iOS and so far I tested beacon monitoring and ranging
Monitoring works fine, ranging however is not giving me any results which it does on Android. So I am going to investigate this now.

**Comment by EricHier** [comment]
That is exactly our problem. The Android version works fine, the iOS version does not range the beacons. Sometimes, after about 20-60 seconds, the beacons are detected but even this does not work reliably.

**Comment by wwwdata** [comment]
Yes, I now also found out that the normal monitoring is also very unreliable and also stops after some time for me

**Comment by EricHier** [comment]
Okay - it's awesome that you can reproduce the problem! :) Do you already have a clue on what is the problem?

**Comment by wwwdata** [comment]
The ranging was never correctly started. As far as I found out it is just calling the same code as for the region monitoring. 
This is the function for foreground ranging. 

Also, it is very important to first register the beacon region before starting the listener. Because the listener is using the previously registered values "automatically". They are essentially living in the memory of the iOS part of the plugin and are then used for the other functions that start-up region/ranging monitoring.

https://github.com/ObjectivityLtd/beacon_monitoring/blob/31d5b1953f7e72e6301d6a217a5eac652980a481/beacon_monitoring/ios/Classes/Dependency/BeaconService.swift#L129-L136

Which is just calling the normal region monitoring code here
https://github.com/ObjectivityLtd/beacon_monitoring/blob/31d5b1953f7e72e6301d6a217a5eac652980a481/beacon_monitoring/ios/Classes/Dependency/BeaconService.swift#L78-L92

This needs to be changed to also take the `Set` of `CLBeaconRegion` and call `startRangingBeacons` for it. Then it was working fine for me and I received data in the ranging stream in Flutter.

I am doing a fork and see what I can fix in the next few days. 

Also, have to thoroughly check all the background monitoring now which for me is the main reason to use this plugin. When I have something I will open a PR here with my fixes. What I also found suspicious is that the plugin also starts to monitor significant location changes which is unnecessary for beacon monitoring:
https://github.com/ObjectivityLtd/beacon_monitoring/blob/1ccfe22a5fe69f6f17f39ea44d08d7c945d4b6cb/beacon_monitoring/ios/Classes/Dependency/LocationService.swift#L47

**Comment by EricHier** [comment]
Oh wow that is awesome! You cannot believe of how much value this is for us! :) Please tell us when you've created the fork. ^^

**Comment by MadDaelim** [comment]
Thank you for your help @wwwdata, once the PR has been created, I will check it as soon as possible and release a new version.

**Comment by wwwdata** [comment]
Opened a PR so far but I am running into way more trouble now when using the background callback.

It looks like this callback is started and then immediately closed again after it is initialized which leaves you with 0 time to do anything :D Trying to figure this out now as well to keep one background isolate running and send all events to it

**Comment by wwwdata** [comment]
also after 1-2 enter and exit regions, it stops working for me.. Maybe it is time to admit I am not able to debug this.
Going to create a small plugin on my own now with way simpler code I guess... Maybe I give it another try tomorrow but to be honest for such a very simple functionality there is A LOT of nontrivial boilerplate code in this repo which makes it very hard and confusing for me to follow.

**Comment by EricHier** [comment]
Okay, I see. :) Switching to a lighter version of the plugin with less functionality would also be an option for us - so please share if you make progress on the new plugin. ^^

**Comment by wwwdata** [comment]
Yes, if my current client allows this and I build a functioning plugin, I will publish it. Also when having less code it is currently very easy for me to debug stuff :) 

- I also made the mistake that my plugin was registered multiple times (I think this is also a current bug in this plugin here). You have to track a singleton instance and only instantiate once inside a sync queue :)
- I noticed that after an `onExit` event, monitoring for that region basically "stops automatically" and the region is removed from the `CLLocationManager` `monitoredRegions`.. Very strange behavior! When only monitoring entry events, it runs indefinitely in my testing so far. Found nothing about this in the iOS documentation.

I will keep on sharing my findings here, maybe that helps you guys out as well. Somehow this topic is "strange" :D
